matrix_3x3_3.0.cpp

```diff
@@ -0,0 +1,343 @@
+			//fprintf(fp, "%f", res.data[row_num][col_num]); ///вопрос: почему в файл записывает НЕ транспонированную матрицу, а исходную
+		}
+		//fprintf(fp, "%c", '\n');
+	}
+	//fprintf(fp, "%c", '\n');
+	return res;
+}
+
+float findDet(Matrix* A)
+{
+	double reduction;
+	float det = 1;
+
+	Matrix res;
+	res.width = A->width;
+	res.height = A->height;
+	for (int row_num = 0; row_num < A->height; row_num++)
+	{
+		for (int col_num = 0; col_num < A->width; col_num++)
+		{
+			res.data[row_num][col_num] = A->data[row_num][col_num];
+		}
+	}
+
+	//приведение матрицы к ступенчатому виду методом гаусса
+	for (int diag_el = 0; diag_el < A->width - 1; diag_el++)
+	{
+		for (int row_num = diag_el + 1; row_num < A->width; row_num++) // i row_num	j col_num	k diag_el
+		{
+			if (res.data[diag_el][diag_el] != 0)
+			{
+				//reduction is ...
+				reduction = - res.data[row_num][diag_el] / res.data[diag_el][diag_el];
+				for (int col_num = 0; col_num < A->width; col_num++)
+				{
+					res.data[row_num][col_num] += res.data[diag_el][col_num] * reduction;
+				}
+			}
+
+		}
+	}
+	//fprintf(fp, "%s\n", "det A:");
+	//det верхнетреугольной матрицы
+	for (int i = 0; i < 3; i++) 
+	{
+		det *= res.data[i][i];
+	}
+	//fprintf(fp, "%f\n", det);
+	//fprintf(fp, "%c", '\n');
+	return det;
+}
+
+Matrix inversion(Matrix* A) //A N
+{
+	double temp;
+	Matrix res;
+	res.width = A->width;
+	res.height = A->height;
+	for (int row_num = 0; row_num < A->height; row_num++)
+	{
+		for (int col_num = 0; col_num < A->width; col_num++)
+		{
+			res.data[row_num][col_num] = A->data[row_num][col_num];
+		}
+	}
+
+	for (int i = 0; i < res.height; i++)
+	{
+		for (int j = 0; j < res.width; j++)
+		{
+			res.data[i][j] = 0.0;
+			if (i == j)
+				res.data[i][j] = 1.0;
+		}
+	}
+		
+	for (int k = 0; k < res.width; k++)
+	{
+		temp = A->data[k][k];
+
+		for (int j = 0; j < res.width; j++)
+		{
+			A->data[k][j] /= temp;
+			res.data[k][j] /= temp;
+		}
+
+		for (int i = k + 1; i < res.width; i++)
+		{
+			temp = A->data[i][k];
+
+			for (int j = 0; j < res.width; j++)
+			{
+				A->data[i][j] -= A->data[k][j] * temp;
+				res.data[i][j] -= res.data[k][j] * temp;
+			}
+		}
+	}
+
+	for (int k = res.width - 1; k > 0; k--)
+	{
+		for (int i = k - 1; i >= 0; i--)
+		{
+			temp = A->data[i][k];
+
+			for (int j = 0; j < res.width; j++)
+			{
+				A->data[i][j] -= A->data[k][j] * temp;
+				res.data[i][j] -= res.data[k][j] * temp;
+			}
+		}
+	}
+
+	for (int i = 0; i < res.width; i++)
+	{
+		for (int j = 0; j < res.width; j++)
+		{
+			A->data[i][j] = res.data[i][j];
+		}
+	}
+
+	//fprintf(fp, "%s\n", "inverse A:");
+	for (int row_num = 0; row_num < res.height; row_num++)
+	{
+		for (int col_num = 0; col_num < res.width; col_num++)
+		{
+			//fprintf(fp, "%f\t", res.data[row_num][col_num]);
+		}
+		//fprintf(fp, "%c", "\n");
+	}
+	//fprintf(fp, "%c", "\n");
+	return res;
+}
+
+int main()
+{
+	Matrix A, B;
+	int width = 3;
+	int height = 3;
+
+	//initialize mat A

```

Comment:
oops. another language in comments... at least do not mix different languages in comments.
the more uniform code you write - more extendable it becomes